In [1]:
%pip install suncalc

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.0.1 -> 26.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip


In [1]:
from calculate_sun_position import calculate_sun_positions_min, create_1_degree_grid_table

In [2]:
sun_position_table = calculate_sun_positions_min()


Total minutes in 2026: 525,600
Above-horizon minutes: 256,820  (48.9%)

Ranges (above-horizon only)
	Azimuth: 0.0237°  →  359.9673°  (span 359.94°)
	Altitude: 2.0002°  →  89.8504°  (span 87.85°)


In [3]:
sun_position_table

{'2026-01-01 03:05': (114.0722, 2.2058),
 '2026-01-01 03:06': (114.1267, 2.4293),
 '2026-01-01 03:07': (114.1815, 2.6527),
 '2026-01-01 03:08': (114.2367, 2.8759),
 '2026-01-01 03:09': (114.2924, 3.0991),
 '2026-01-01 03:10': (114.3484, 3.3221),
 '2026-01-01 03:11': (114.4049, 3.5451),
 '2026-01-01 03:12': (114.4618, 3.768),
 '2026-01-01 03:13': (114.519, 3.9907),
 '2026-01-01 03:14': (114.5767, 4.2134),
 '2026-01-01 03:15': (114.6348, 4.436),
 '2026-01-01 03:16': (114.6933, 4.6584),
 '2026-01-01 03:17': (114.7523, 4.8808),
 '2026-01-01 03:18': (114.8116, 5.103),
 '2026-01-01 03:19': (114.8714, 5.3252),
 '2026-01-01 03:20': (114.9316, 5.5472),
 '2026-01-01 03:21': (114.9923, 5.7691),
 '2026-01-01 03:22': (115.0533, 5.9909),
 '2026-01-01 03:23': (115.1148, 6.2126),
 '2026-01-01 03:24': (115.1767, 6.4342),
 '2026-01-01 03:25': (115.2391, 6.6557),
 '2026-01-01 03:26': (115.3019, 6.8771),
 '2026-01-01 03:27': (115.3651, 7.0983),
 '2026-01-01 03:28': (115.4288, 7.3194),
 '2026-01-01 03:29':

In [17]:
sun_position_table

{'2026-01-01 02:56': (113.6, 0.1906),
 '2026-01-01 02:57': (113.6509, 0.4149),
 '2026-01-01 02:58': (113.7022, 0.6391),
 '2026-01-01 02:59': (113.7539, 0.8632),
 '2026-01-01 03:00': (113.806, 1.0872),
 '2026-01-01 03:01': (113.8584, 1.3111),
 '2026-01-01 03:02': (113.9113, 1.5349),
 '2026-01-01 03:03': (113.9645, 1.7587),
 '2026-01-01 03:04': (114.0182, 1.9823),
 '2026-01-01 03:05': (114.0722, 2.2058),
 '2026-01-01 03:06': (114.1267, 2.4293),
 '2026-01-01 03:07': (114.1815, 2.6527),
 '2026-01-01 03:08': (114.2367, 2.8759),
 '2026-01-01 03:09': (114.2924, 3.0991),
 '2026-01-01 03:10': (114.3484, 3.3221),
 '2026-01-01 03:11': (114.4049, 3.5451),
 '2026-01-01 03:12': (114.4618, 3.768),
 '2026-01-01 03:13': (114.519, 3.9907),
 '2026-01-01 03:14': (114.5767, 4.2134),
 '2026-01-01 03:15': (114.6348, 4.436),
 '2026-01-01 03:16': (114.6933, 4.6584),
 '2026-01-01 03:17': (114.7523, 4.8808),
 '2026-01-01 03:18': (114.8116, 5.103),
 '2026-01-01 03:19': (114.8714, 5.3252),
 '2026-01-01 03:20': (11

In [4]:
grid_sun_position_table = create_1_degree_grid_table(sun_position_table)


Number of cells: 16252


In [14]:
import pickle

list_grid_pos = list(grid_sun_position_table.keys())

with open("grid_sun_pos.pkl", "wb") as f:
    pickle.dump(list_grid_pos, f)

In [6]:
import pickle

with open("grid_sun_pos.pkl", "rb") as f:
    list_grid = pickle.load(f)

In [7]:
list_grid

[(114, 2),
 (114, 3),
 (114, 4),
 (115, 4),
 (115, 5),
 (115, 6),
 (115, 7),
 (115, 8),
 (116, 8),
 (116, 9),
 (116, 10),
 (116, 11),
 (117, 11),
 (117, 12),
 (117, 13),
 (117, 14),
 (118, 14),
 (118, 15),
 (118, 16),
 (119, 16),
 (119, 17),
 (119, 18),
 (119, 19),
 (120, 19),
 (120, 20),
 (120, 21),
 (121, 21),
 (121, 22),
 (121, 23),
 (122, 23),
 (122, 24),
 (122, 25),
 (123, 25),
 (123, 26),
 (123, 27),
 (124, 27),
 (124, 28),
 (125, 29),
 (125, 30),
 (126, 30),
 (126, 31),
 (127, 32),
 (127, 33),
 (128, 33),
 (128, 34),
 (129, 34),
 (129, 35),
 (130, 35),
 (130, 36),
 (131, 37),
 (132, 38),
 (132, 39),
 (133, 39),
 (134, 40),
 (135, 41),
 (136, 41),
 (136, 42),
 (137, 42),
 (137, 43),
 (138, 43),
 (138, 44),
 (139, 44),
 (140, 44),
 (140, 45),
 (141, 45),
 (141, 46),
 (142, 46),
 (143, 46),
 (143, 47),
 (144, 47),
 (145, 47),
 (145, 48),
 (146, 48),
 (147, 49),
 (148, 49),
 (149, 49),
 (149, 50),
 (150, 50),
 (151, 50),
 (151, 51),
 (152, 51),
 (153, 51),
 (154, 51),
 (154, 52),
 (

LAT  = 52.13896498167249
LON  = 11.645601397328756
Total minutes in 2026  : 525,600
Above-horizon minutes  : 263,054  (50.0%)

Ranges (above-horizon only)
  Azimuth  : 0.0237°  →  359.9673°  (span 359.94°)
  Altitude : 0.0012°  →  89.8504°  (span 89.85°)




LAT  = 52.13928
LON  = 11.65027
Total minutes in 2026  : 525,600
Above-horizon minutes  : 263,052  (50.0%)

Ranges (above-horizon only)
  Azimuth  : 0.0219°  →  359.9659°  (span 359.94°)
  Altitude : 0.0006°  →  89.8536°  (span 89.85°)